agora-scripts/ScriptExport/API.hs

```diff
@@ -0,0 +1,110 @@
+{- | Module   : ScriptExport.API
+     Maintainer : emi@haskell.fyi
+     Description: API for script exporter.
+
+     API for script exporter.
+-}
+module ScriptExport.API (
+  API,
+  runServer,
+) where
+
+import Codec.Serialise.Orphans ()
+import Data.Aeson qualified as Aeson
+import Data.Cache.Cached (cachedForM)
+import Data.Function ((&))
+import Data.Proxy (Proxy (Proxy))
+import Data.Text (Text)
+import GHC.Generics qualified as GHC
+import Network.HTTP.Types qualified as Http
+import Network.Wai qualified as Wai
+import Network.Wai.Handler.Warp qualified as Warp
+import Network.Wai.Middleware.Cors (CorsResourcePolicy (corsRequestHeaders), cors, simpleCorsResourcePolicy)
+import Prettyprinter (Pretty (pretty), defaultLayoutOptions, hsep, layoutPretty, viaShow)
+import Prettyprinter.Render.String (renderString)
+import ScriptExport.Options (Options (..))
+import ScriptExport.Types (Builders, ScriptQuery (ScriptQuery), runQuery)
+import ScriptExport.Types qualified as Builders
+import Servant.API (Capture, Get, JSON, Post, ReqBody, (:<|>) (..), type (:>))
+import Servant.Server qualified as Servant
+import System.Clock (TimeSpec (TimeSpec))
+import Text.Printf (printf)
+
+{- | Servant API type for script generation.
+
+     @since 0.2.0
+-}
+type API =
+  -- /query-script/:name
+  "query-script"
+    :> Capture "name" Text
+    :> ReqBody '[JSON] Aeson.Value
+    :> Post '[JSON] Aeson.Value
```

Comment:
I still believe that we should use `GET` instead of `POST` here, cause this endpoint doesn't cause any side effects on the server.

Reply:
HTTP is weird about how things should work. A body on a `GET` method must not be used for the response. It's weird but that's how things are.